### PR TITLE
replaced "title" with "id" for ColBERT document identifier

### PR DESCRIPTION
**Pull request by epinzur**
ColBERT should not require a `title` for each document. It should be an optional `id`, that is generated if not passed.

If a developer wants to save the `filename` or other `title` for a document, they should do it in the metadata.

This PR is in place of https://github.com/datastax/ragstack-ai/pull/338

Also made a bunch of other name changes.

### CODE REVIEW

**Review by nicoloboschi — APPROVED** [review]

**Review by zzzming — APPROVED** [review]
one minor comment. We don't need parent_id in the PerTokenEmbeddings.

File: ragstack/colbert/colbert_embedding.py [hunk 1]
```diff
@@ -123,17 +123,15 @@ def __init__(
             self.checkpoint = self.checkpoint.cuda()
 
     def embed_documents(
```
Comment (nicoloboschi) [inline ragstack/colbert/colbert_embedding.py:125]: is "document" correct here? 
if so, `embed_query` should be embed_document

Probably we should use embed_texts and embed_text

**Review by nicoloboschi — APPROVED** [review]

File: ragstack/colbert/colbert_embedding.py (hunk 1 above)
Comment (epinzur) [reply ragstack/colbert/colbert_embedding.py:125]: this is just for implementing the abstract method in this class: https://github.com/datastax/ragstack-ai/pull/347/files#diff-c553042020c3e53b973244f9a67fb153ece342591f1ed32b563f97ce7f2ff0faR76

Which I'm not sure if we need. What do you think?

Comment (nicoloboschi) [reply ragstack/colbert/colbert_embedding.py:125]: yes let's keep the interface, code is much leaner to read. Wonder if we want to rename the interface methods too

**Pull request merged**